src/app/app.module.ts

```diff
@@ -54,11 +55,17 @@ const appRoutes: Routes = [
     path: 'profile',
     component: ProfileComponent,
     canActivate: [ OktaAuthGuard ],
+    data: {
+      onAuthRequired: onNeedsAuthGuard
+    }
```

Comment:
@shuowu Can you confirm that this new `data` block is necessary? When I remove it, using the Sign-In Widget doesn't work. It redirects me to Okta and makes me log in again. With it, things work as expected. I ask because this doesn't seem to be documented in the [migrating guide](https://github.com/okta/okta-angular/blob/okta-angular-4.0.0/MIGRATING.md).

Reply:
`onAuthRequired` from the data field set the guard per route. You can also set it globally from OKTA_CONFIG value (around line 87)